### PR TITLE
Future returning signatures

### PR DESCRIPTION
**Pull request by vietj**

### CODE REVIEW

File: src/test/java/io/vertx/serviceproxy/codegen/future/FuturizedProxy.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+/**
+ * @author <a href="http://tfox.org">Tim Fox</a>
+ */
+@ProxyGen
+public interface FuturizedProxy {
+
+  Future<String> future0();
+
+/*
```
Comment (slinkydeveloper) [inline src/test/java/io/vertx/serviceproxy/codegen/future/FuturizedProxy.java:14]: Can you remove these comments below?

File: src/test/java/io/vertx/serviceproxy/test/FuturizedServiceProxyTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * @author <a href="http://tfox.org">Tim Fox</a>
+ * @author lalitrao
```
Comment (slinkydeveloper) [inline src/test/java/io/vertx/serviceproxy/test/FuturizedServiceProxyTest.java:52]: ?!

**Review by slinkydeveloper — COMMENTED** [review]
I think this is looking good

File: src/test/java/io/vertx/serviceproxy/codegen/future/FuturizedProxy.java (hunk 1 above)
Comment (vietj) [reply src/test/java/io/vertx/serviceproxy/codegen/future/FuturizedProxy.java:14]: I will

File: src/test/java/io/vertx/serviceproxy/test/FuturizedServiceProxyTest.java (hunk 2 above)
Comment (vietj) [reply src/test/java/io/vertx/serviceproxy/test/FuturizedServiceProxyTest.java:52]: I'll remove it :-)

File: src/main/java/io/vertx/serviceproxy/generator/ServiceProxyGen.java [hunk 3]
```diff
@@ -110,8 +111,8 @@ private void generateMethods(ProxyModel model, CodeWriter writer) {
 
   private void generateMethodBody(ProxyMethodInfo method, CodeWriter writer) {
     ParamInfo lastParam = !method.getParams().isEmpty() ? method.getParam(method.getParams().size() - 1) : null;
-    boolean hasResultHandler = utils.isResultHandler(lastParam);
-    if (hasResultHandler) {
+//    boolean hasResultHandler = utils.isResultHandler(lastParam);
```
Comment (vietj) [inline src/main/java/io/vertx/serviceproxy/generator/ServiceProxyGen.java:114]: need to remove that

**Comment by vietj** [comment]
@slinkydeveloper I removed the comments and updated the documentation to use futures.

**Pull request merged**